frontend/src/screens/Edition/ContentView.tsx

```diff
@@ -0,0 +1,160 @@
+    if (!selectedFile) {
+      return <p>No file selected.</p>;
+    }
+
+    const fileDetails = renderFileDetails();
+    switch (selectedFile.extension) {
+      case 'txt':
+        return <TextFileComponent file={selectedFile} onSave={handleSave} />;
+
+      case 'md':
+        return <MarkdownFile file={selectedFile} onSave={handleSave} />;
+      case 'jpg':
+      case 'jpeg':
+      case 'gif':
+      case 'png':
+        return (
+          <img
+            src={`data:image/${selectedFile.extension};base64,${selectedFile.content}`}
+            alt={selectedFile.name}
+            style={{ maxWidth: '100%' }}
+          />
+        );
+      case 'mp4':
+        return (
+          <video
+            src={`data:video/${selectedFile.extension};base64,${selectedFile.content}`}
+            controls
+            style={{ maxWidth: '100%' }}
+          />
+        );
+      case 'pdf': {
+        const pdfDataUrl = `data:application/pdf;base64,${selectedFile.content}`;
+        return (
+          <object
+            data={pdfDataUrl}
+            type="application/pdf"
+            width="100%"
+            height="500px"
+            style={{ minHeight: '500px' }}>
+            {' '}
+            PDF cannot be displayed, download it{' '}
+            <a href={pdfDataUrl} download={`${selectedFile.name}.pdf`}>
+              here
+            </a>
+            .
+          </object>
+        );
+      }
+      default:
+        return (
+          <p>
+            Unsupported file type.{' '}
+            <a href={`data:text/plain;base64,${selectedFile.content}`} download={selectedFile.name}>
+              Download file
+            </a>
+          </p>
+        );
+    }
+  };
+
+  return (
+    <Grid celled="internally">
+      {selectedFile && (
+        <Grid.Row>
+          <Grid.Column width={16}>
+            <Accordion fluid styled>
+              <Accordion.Title active={activeIndex === 0} onClick={handleClick}>
+                <Icon name="dropdown" />
+                {selectedFile.name}.{selectedFile.extension}
+              </Accordion.Title>
+              <Accordion.Content active={activeIndex === 0}>
+                <p>
+                  <strong>Can be edited:</strong> {selectedFile.can_be_edited ? 'Yes' : 'No'}
+                </p>
+                <p>
+                  <strong>Can be previewed:</strong> {selectedFile.can_be_previewed ? 'Yes' : 'No'}
+                </p>
+                <p>
+                  <strong>Is attachment:</strong> {selectedFile.is_attachment ? 'Yes' : 'No'}
+                </p>
+              </Accordion.Content>
+            </Accordion>
+          </Grid.Column>
+        </Grid.Row>
+      )}
+      <Grid.Row>
+        <Grid.Column width={16}>
+          <Segment>
+            {selectedFile ? renderFileContent() : <p>Select a file to view its content.</p>}
+          </Segment>
+        </Grid.Column>
+      </Grid.Row>
+    </Grid>
+  );
+};
+
+export default ContentView;
```

Comment:
pretty big file - probably should be separated

Reply:
Yes, dropdown will be separate file in the future. Same goes for components designated to render different filetypes